### PR TITLE
feat: bitly list active links

### PR DESCRIPTION
**Pull request by FlorentLvr**

### CODE REVIEW

**Comment by Dr0p42** [comment]
No linked issues found. Please add the corresponding issues in the pull request description. <br/> [Use GitHub automation to close the issue when a PR is merged](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

**Review by srini047 — COMMENTED** [review]
@FlorentLvr The code looks fine to me. But I cannot generate a Bitly access token to verify explicitly. Other than that we are good to go.

**Comment by FlorentLvr** [comment]
> @FlorentLvr The code looks fine to me. But I cannot generate a Bitly access token to verify explicitly. Other than that we are good to go.

Hi Srini, there is a free trial on Bitly: https://bitly.com/pages/landing/free-plan
You should normally be able to create an access token :). 
Can you try it please? 🙏

**Comment by srini047** [comment]
> > @FlorentLvr The code looks fine to me. But I cannot generate a Bitly access token to verify explicitly. Other than that we are good to go.
> 
> Hi Srini, there is a free trial on Bitly: https://bitly.com/pages/landing/free-plan You should normally be able to create an access token :). Can you try it please? 🙏

@FlorentLvr As mentioned I did create a free account and then landed here.

Screenshot 2023-02-20 at 7 10 20 PM

When I click generate token it says password is required. But I signed up using Google OAuth. Any help to generate an access token is appreciated.

**Comment by FlorentLvr** [comment]
> > > @FlorentLvr The code looks fine to me. But I cannot generate a Bitly access token to verify explicitly. Other than that we are good to go.
> > 
> > 
> > Hi Srini, there is a free trial on Bitly: https://bitly.com/pages/landing/free-plan You should normally be able to create an access token :). Can you try it please? 🙏
> 
> @FlorentLvr As mentioned I did create a free account and then landed here.
> 
> Screenshot 2023-02-20 at 7 10 20 PM
> 
> When I click generate token it says password is required. But I signed up using Google OAuth. Any help to generate an access token is appreciated.

Hi @srini047! I just try it as well and I had to reset my password manualy to generate the token. Can you try it as well and let me know if it works?

https://support.bitly.com/hc/en-us/articles/230647907-How-do-I-generate-an-OAuth-access-token-for-the-Bitly-API-
image

File: Bitly/Bitly_List_Active_Links.ipynb [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "46436a7b-1379-4e7e-a0b4-967126e7bcbd",
+   "metadata": {
+    "papermill": {},
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "active_links = list_active_links(token)\n",
```
Comment (srini047) [inline Bitly/Bitly_List_Active_Links.ipynb:217]: Screenshot 2023-02-21 at 1 41 29 PM

Getting `HTTPError: 404 Client Error: Not Found for url: https://api-ssl.bitly.com/v4/user/active_links?size=100` while running the notebook.

**Review by srini047 — COMMENTED** [review]
@FlorentLvr I get HTTPError while trying to call the function. Please review it

**Comment by FlorentLvr** [comment]
@srini047, regarding the api documentation it does not seem possible to get all active links in Bitly. Do you agree?
So, I just change the template to "retrieve bitlink". Can you check if it is working on your side?

File: Bitly/Bitly_Retrieve_Bitlink.ipynb [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    "papermill": {},
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def retrieve_bitlink(token, bitlink):\n",
+    "    headers = {\"Authorization\": f\"Bearer {token}\"}\n",
+    "    url = f\"https://api-ssl.bitly.com/v4/bitlinks/{bitlink}\"\n",
+    "    response = requests.get(url, headers=headers)\n",
+    "    response.raise_for_status()\n",
```
Comment (srini047) [inline Bitly/Bitly_Retrieve_Bitlink.ipynb:172]: @FlorentLvr Getting an HTTPError here.

Screenshot 2023-02-24 at 7 04 33 PM

**Review by srini047 — COMMENTED** [review]
@FlorentLvr Facing errors as stated. Need to look for possible other possible solutions.

File: Bitly/Bitly_Retrieve_Bitlink.ipynb (hunk 2 above)
Comment (FlorentLvr) [reply Bitly/Bitly_Retrieve_Bitlink.ipynb:172]: @srini047, it seems you are using the link in demo that why it is not working. You have to use a bitlink from your account. :)

**Pull request merged**